### ISSUE TITLE
RBioFormats library needs internet connection to load

### ISSUE THREAD

**Issue by tvazm** [aoles/RBioFormats#23]
Hello,

I was using RBioFormats without any problems until I noticed that I could not load RBioFormats library when my computer is offline. 

I get the error message (in RStudio):

> library(RBioFormats)
Error: package or namespace load failed for ‘RBioFormats’:
 .onLoad failed in loadNamespace() for 'RBioFormats', details:
  call: NULL
  error: failed to download Bio-Formats Java library.
  Check your internet connection and try again.
In addition: Warning message:
In file(con, "r") :
  InternetOpenUrl failed: 'The server name or address could not be resolved'

I am running Windows
> sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18362)

I installed RBioformats (on August 2020) doing
devtools::install_github("aoles/RBioFormats", INSTALL_opts=c("--no-multiarch"))

and it works normally when I am connected to Internet.

I don't really know what to do, but from what I read here, I thought that perhaps I needed to set a classpath to the bioformats_package.jar for rJava.  So I wrote (RStudio):

> a<-.libPaths()
> ab<-paste0(a,'/RBioFormats/Java')
> .jaddClassPath(ab)
> .jclassPath()
[1] "C:\\Program Files\\R\\R-4.0.2\\library\\rJava\\java"      
[2] "C:\\Program Files\\Portugal Identity Card"                
[3] "C:\\Program Files\\R\\R-4.0.2\\library\\RBioFormats\\Java"

However, this didn't work at all, since I get the same error when I am offline. 

I can confirm that now the bioformats_package.jar is in the rJava classpath 

> .jclassPath()
[1] "C:\\Program Files\\R\\R-4.0.2\\library\\rJava\\java"      
[2] "C:\\Program Files\\Portugal Identity Card"                
[3] "C:\\Program Files\\R\\R-4.0.2\\library\\RBioFormats\\Java"

And that the bioformats_package.jar is on the 'ab' folder
> ab
[1] "C:/Program Files/R/R-4.0.2/library/RBioFormats/Java"
> setwd(ab)
> list.files()
[1] "bioformats_package.jar" "RBioFormats.jar"

Still, it doesn't work. I don't really understand what is happening, but I need to be able to load the library without Internet connection. I googled it and I know that other people have (or have had) the same problem, but I cannot find any solution. Could you please help?

Thank you.

Teresa

**Comment by aoles** [comment]
Dear Teresa,

Thanks for your feedback! The issue originates from the fact that on load the package performs a check against a remote resource. When implementing this solution I overlooked the potential consequences when internet connection is not available. I'm sorry for the inconvenience. 

I will look into fixing this ASAP, so stay tuned.

Cheers, 
Andrzej

**Comment by tvazm** [comment]
Thank you Andrzej, I will!

**Comment by aoles** [comment]
Dear Teresa,

I have just updated the package. Please try reinstalling the package and let me know if the fix solves your issue.

Cheers,
Andrzej

**Comment by tvazm** [comment]
Dear Andrzej,

I reinstalled it and it works, it loads without Internet now. After installation, I turned off Internet and it loaded. I also rebooted the computer without Internet and it still loaded and worked without any problem. 

So my problem has been solved. Thank you very much! 

Teresa

For information:
I reinstalled doing:
devtools::install_github("aoles/RBioFormats", INSTALL_opts=c("--no-multiarch"))

I am running Windows:
> sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18362)
